File: Data Structures & Algorithms/majority-element/submission-0.java
class Solution {
    public int majorityElement(int[] nums) {
        TreeMap<Integer, Integer> map = new TreeMap<>();
        for(int i = 0; i< nums.length; i++){
            if(map.containsKey(nums[i]))
            map.put(nums[i], map.get(nums[i]) + 1);
            else{
               map.put(nums[i], 1); 
            }
        }
        int max = 0;
        int maxElem = 0;
        for(int i : map.keySet()){
            System.out.println(i);
            if(max <= map.get(i)){
                max = map.get(i);
                maxElem = i;
            }
        }
        return maxElem;
    }
}

File: Data Structures & Algorithms/search-insert-position/submission-0.cpp
class Solution {
public:
    int searchInsert(vector<int>& nums, int target) {
        int l = 0;
        int r = nums.size();

        while(l < r){
            int m = (r - l)/2 + l;
            if(nums[m] == target){ 
                return m;
            }
            if(nums[m] > target){
                r = m;
            }else{
                l = m + 1;
            }
        } 
        return r;
    }
};